src/main/java/com/doddi/vulnerability_scanner_idea_plugin/system/golang/GolangPopulator.java

```diff
@@ -10,9 +9,15 @@
 import com.doddi.vulnerability_scanner_idea_plugin.entity.LibraryToScan;
 import com.doddi.vulnerability_scanner_idea_plugin.entity.VulnerableComponent;
 import com.doddi.vulnerability_scanner_idea_plugin.system.Populator;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.project.Project;
 
-public class GolangPopulator implements Populator {
-    private final Map<String, VulnerableComponent> components = new HashMap<>();
+public class GolangPopulator
+    extends Populator
+{
+    static public GolangPopulator getInstance(final Project project) {
```

Comment:
These getInstance methods in the populators are odd. They're static so you have to know the class you're going to call it from so why wouldn't a caller just use ```ServiceManager.getService(project, GolangPopulator.class);```

Reply:
It is just a wrapper, and it seems to be a common way of wrapping. It is even part of the default template when creating services under the intellij plugin
<img width="444" alt="Screenshot 2020-06-29 at 16 34 20" src="https://user-images.githubusercontent.com/260767/86025734-73a5f100-ba26-11ea-9167-a9feb3b0fe58.png">